plugins/backend-trace-forwarder/src/Cardano/BM/Backend/TraceForwarder.lhs

```diff
@@ -144,16 +144,30 @@ instance (ToJSON a) => IsEffectuator TraceForwarder a where
 
       writeMessageToQueue currentTF' = do
         let queue = tfQueue currentTF'
+        currentQueueSize <- atomically $ TBQ.lengthTBQueue queue
         noCapacity <- atomically $ TBQ.isFullTBQueue queue
```

Comment:
This pair of independent actions could return inconsistent results.

The more correct version would be:

```suggestion
        (,) currentQueueSize noCapacity <- atomically $
            (,) <$>TBQ.lengthTBQueue queue
                <*> TBQ.isFullTBQueue queue
```